webui/react/src/pages/TaskList.tsx

```diff
@@ -249,6 +104,30 @@ const TaskList: React.FC = () => {
     return false;
   }, [ selectedTasks ]);
 
+  const filterConfigMap = useMemo(() => {
+    return filterKeys.reduce((acc, key) => {
+      const config = settingsConfig.settings.find(config => config.key === key);
+      return { ...acc, [key]: config };
+    }, {} as Record<keyof Settings, SettingsConfigProp>);
+  }, []);
+
+  const filterCount = useMemo(() => {
+    return filterKeys.reduce((acc, key) => {
+      const config = filterConfigMap[key];
+      const settingsValue = settings[key];
+      const settingsIsSet = settingsValue && !isEqual(settingsValue, config.defaultValue);
+      return acc + (settingsIsSet ? 1 : 0);
+    }, 0);
+  }, [ filterConfigMap, settings ]);
+
+  const resetFilters = useCallback(() => {
```

Comment:
suggestion: I recall you had a resetFilters in the useSettings hook. if that was a resetSettings that optionally took a set of setting keys? would that  make it reusable here?

Reply:
this is a great idea!